app/models/role.rb

```diff
@@ -242,4 +244,15 @@ def prevent_changes
 
     raise ActiveRecord::ReadOnlyRecord unless new_record? || only_archival
   end
+
+  def set_last_active_role
+    still_has_active_role = person.roles.with_deleted.where(roles: { deleted_at: nil }).present?
```

Comment:
do we need to use the `with_deleted` scope here? 
doesn't the default scope for roles already return only those where `deleted_at: nil` ?

Reply:
Nope, returns `deleted_at IS NULL OR deleted_at >= :now`